vault/resource_gcp_secret_static_account_test.go

```diff
@@ -0,0 +1,397 @@
+					resource.TestCheckResourceAttr("vault_gcp_secret_backend.test", "path", backend),
+					resource.TestCheckResourceAttr("vault_gcp_secret_static_account.test", "backend", backend),
+					resource.TestCheckResourceAttr("vault_gcp_secret_static_account.test", "static_account", staticAccount),
+					resource.TestCheckResourceAttr("vault_gcp_secret_static_account.test", "secret_type", "access_token"),
+					resource.TestCheckResourceAttr("vault_gcp_secret_static_account.test", "service_account_email", serviceAccountEmail),
+					resource.TestCheckResourceAttr("vault_gcp_secret_static_account.test", "service_account_project", project),
+					resource.TestCheckResourceAttr("vault_gcp_secret_static_account.test", "token_scopes.#", "1"),
+					resource.TestCheckResourceAttr("vault_gcp_secret_static_account.test", "token_scopes.2400041053", "https://www.googleapis.com/auth/cloud-platform"),
+					resource.TestCheckResourceAttr("vault_gcp_secret_static_account.test", "binding.#", "1"),
+					resource.TestCheckResourceAttr("vault_gcp_secret_static_account.test", fmt.Sprintf("binding.%d.resource", updatedHash), fmt.Sprintf("//cloudresourcemanager.googleapis.com/projects/%s", project)),
+					resource.TestCheckResourceAttr("vault_gcp_secret_static_account.test", fmt.Sprintf("binding.%d.roles.#", updatedHash), "1"),
+					resource.TestCheckResourceAttr("vault_gcp_secret_static_account.test", fmt.Sprintf("binding.%d.roles.2133424675", updatedHash), updatedRole),
+				),
+			},
+			{
+				Config: keyConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testGCPSecretStaticAccount_attrs(backend, staticAccount),
+					resource.TestCheckResourceAttr("vault_gcp_secret_backend.test", "path", backend),
+					resource.TestCheckResourceAttr("vault_gcp_secret_static_account.test", "backend", backend),
+					resource.TestCheckResourceAttr("vault_gcp_secret_static_account.test", "static_account", staticAccount),
+					resource.TestCheckResourceAttr("vault_gcp_secret_static_account.test", "secret_type", "service_account_key"),
+					resource.TestCheckResourceAttr("vault_gcp_secret_static_account.test", "service_account_email", serviceAccountEmail),
+					resource.TestCheckResourceAttr("vault_gcp_secret_static_account.test", "service_account_project", project),
+					resource.TestCheckResourceAttr("vault_gcp_secret_static_account.test", "binding.#", "1"),
+					resource.TestCheckResourceAttr("vault_gcp_secret_static_account.test", fmt.Sprintf("binding.%d.resource", keyHash), fmt.Sprintf("//cloudresourcemanager.googleapis.com/projects/%s", project)),
+					resource.TestCheckResourceAttr("vault_gcp_secret_static_account.test", fmt.Sprintf("binding.%d.roles.#", keyHash), "1"),
+					resource.TestCheckResourceAttr("vault_gcp_secret_static_account.test", fmt.Sprintf("binding.%d.roles.2133424675", keyHash), updatedRole),
+				),
+			},
+		},
+	})
+}
+
+func testGCPSecretStaticAccount_attrs(backend, staticAccount string) resource.TestCheckFunc {
```

Comment:
I wonder if there is way we could perhaps refactor this test function to other, more discrete functions. As-is, it very thorough, but rather challenging to consume in on go :)

Reply:
This "pattern" is used throughout the tests in the provider which I mimicked here.. Perhaps a cleanup/refactor PR further down the line?

Reply:
Okay, that sounds fine, since a precedent has already been set.